module_evaluations/TCR-31_2023-09-06-mod-lists.MD

```diff
@@ -0,0 +1,69 @@
+* [x] Unit tests have 80% coverage or greater, and are based on [officially approved technologies](https://wiki.folio.org/display/TC/Officially+Supported+Technologies)
+  * _Note: 89.6% See [Sonarcloud](https://sonarcloud.io/summary/overall?id=org.folio%3Amod-lists)._
+
+## Backend
+* [x] Module's repository includes a compliant Module Descriptor
+  * _Note: read more at https://github.com/folio-org/okapi/blob/master/okapi-core/src/main/raml/ModuleDescriptor.json_
+* [X] Module includes executable implementations of all endpoints in the provides section of the Module Descriptor
+  * _Note: all endpoints in descriptor have entries in `list.yaml`, all `operationId` values listed in `list.yaml` have corresponding methods with matching arguments in `ListController.java`, `ListExportController.java`, and `ListConfigurationController.java`._
+* [x] Environment vars are documented in the ModuleDescriptor
+  * _Note: read more at [https://wiki.folio.org/pages/viewpage.action?pageId=65110683](https://wiki.folio.org/pages/viewpage.action?pageId=65110683)_
+* [x] If a module provides interfaces intended to be consumed by other FOLIO Modules, they must be defined in the Module Descriptor "provides" section
+* [x] All API endpoints are documented in RAML or OpenAPI
+* [x] All API endpoints protected with appropriate permissions as per the following guidelines and recommendations, e.g. avoid using `*.all` permissions, all necessary module permissions are assigned, etc.
+  * _Note: read more at https://dev.folio.org/guidelines/naming-conventions/ and https://wiki.folio.org/display/DD/Permission+Set+Guidelines_
+* [x] ~Module provides reference data (if applicable), e.g. if there is a controlled vocabulary where the module requires at least one value~
+* [x] ~If provided, integration (API) tests must be written in an [officially approved technology](https://wiki.folio.org/display/TC/Officially+Supported+Technologies)~
+  * _Note: while it's strongly recommended that modules implement integration tests, it's not a requirement_
+  * _Note: these tests are defined in https://github.com/folio-org/folio-integration-tests_
+* [x] Data is segregated by tenant at the storage layer
+* [x] The module doesn't access data in DB schemas other than its own and public
+* [x] The module responds with a tenant's content based on x-okapi-tenant header
+* [x] Standard GET `/admin/health` endpoint returning a 200 response
+  * _Note: read more at https://wiki.folio.org/display/DD/Back+End+Module+Health+Check+Protocol_
+  * _Note: yes, via actuator_
+* [x] High Availability (HA) compliant
+  * Possible red flags:
+    * Connection affinity / sticky sessions / etc. are used
+    * Local container storage is used
+    * Services are stateful
+* [ ] Module only uses infrastructure / platform technologies on the [officially approved technologies](https://wiki.folio.org/display/TC/Officially+Supported+Technologies) list.
```

Comment:
Which technology is used that isn't considered part of the list?

Is it min.io?

Reply:
Right, Minio isn't actually on a list and [DR-000026 - S3/MinIO](https://wiki.folio.org/pages/viewpage.action?pageId=96419874) has been sitting in draft for a long time. @craigmcnally and I agree that while TC is probably fine with this, we really need to discuss make an explicit decision for this TCR, as will as prod the DR along.

Reply:
Practically, it is on the list because we've got modules already in use that use it